starters/next-drupal-starter/components/article.js

```diff
@@ -0,0 +1,34 @@
+import { IMAGE_URL } from "../lib/constants";
+import Image from "next/image";
+import Link from "next/link";
+
+export default function Article({ article }) {
```

Comment:
I see what you're saying about assumptions about the shape of the data. This example is pretty simple so it isn't the end of the world, but it still illustrates the problem.

In the past I've addressed this by making components like this as presentational as possible and accepting the exact props they need rather than a CMS specific object. In this example the props would be: imgSrc, title, body. That makes this an article as in `<article>` not `node--article`. That would allow someone to use this article component even if their field was named something like `field_hero_image` In fact, it is so generic that this component be used with another CMS and/or abstracted into something like the 'react-kit` package we've considered previously.

Something still needs to handle the logic to prepare the data for this presentational component. I've used a provider-style component for that in the past. Simplified example:

```js
export default function ArticleProvider({ node }) {
  const imgSrc = article?.field_media_image?.field_media_image?.uri?.url || "";
  // handle the other props here
    return (
       <Article imgSrc={imgSrc} />
    )
}
```

Maybe that would be better expressed as a hook these days.

```js
const {imgSrc, title, body} = useArticle(node);

<Article
  imgSrc={imgSrc}
  title={title}
  etc...
/>
```

Details to work out for sure especially as we start thinking about the provider or hook, but that is the basic idea. Thoughts?

Reply:
Ah yeah that makes more sense! I like the idea of a hook or HOC. Might pull the grid out into an HOC also.
Will also be more explicit with the props that are coming back from the `get*Props` functions.